Sprinkle of Yeezus/Settings.swift

```diff
@@ -11,52 +11,68 @@ import UIKit
 import UserNotifications
 
 class SettingsPage: UIViewController {
-    
 
-    @IBOutlet weak var InfoLabel: UILabel!
-    @IBOutlet weak var DescButton: UIButton!
-    @IBOutlet weak var DescLabel: UILabel!
-    @IBOutlet weak var Switch: UISwitch!
+    @IBOutlet private weak var InfoLabel: UILabel!
+    @IBOutlet private weak var DescButton: UIButton!
+    @IBOutlet private weak var DescLabel: UILabel!
+    @IBOutlet private weak var Switch: UISwitch!
+    @IBOutlet private weak var TimePicked: UIDatePicker!
+    
+    private var sprinkleTimePicked = Date()
+    private let notificationCenter = UNUserNotificationCenter.current()
     
-    var notificationsOn = Bool()
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        sprinkleNotifications(sprinkleList)
+    }
     
-    @IBAction func NotificationSwitch(_ sender: UISwitch) {
-        if Switch.isOn == true{
-            UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .sound], completionHandler: {didAllow, error in})
-            
-            DescLabel.isHidden = false
-            DescButton.isHidden = false
-            
-            notificationsOn = true
-        } else {
-            DescLabel.isHidden = true
-            DescButton.isHidden = true
-            notificationsOn = false
+    // MARK: - IBAction
+    @IBAction private func NotificationSwitch(_ sender: UISwitch) {
+        if sender.isOn {
+            notificationCenter.requestAuthorization(options: [.alert, .sound], completionHandler: { didAllow, error in
+                DispatchQueue.main.async {
+                    sender.isOn = didAllow
+                    self.updateUI()
+                }
+            })
         }
+        
+        updateUI()
     }
     
-    @IBOutlet weak var TimePicked: UIDatePicker!
-    var sprinkleTimePicked = Date()
-    @IBAction func TimePicker(_ sender: UIDatePicker){
+    @IBAction private func TimePicker(_ sender: UIDatePicker){
         sprinkleTimePicked = TimePicked.date
-    }
+    }  
     
-    func sprinkleNotifications(_ sprinkleList: Array<Sprinkle>) {
-        if notificationsOn == true{
-            
-            let notification = UNMutableNotificationContent()
-            let notificationText = pickRandomQuote(sprinkleList)
-            notification.body = " \"\(notificationText.quote)\" \(notificationText.quoteSource), \(notificationText.date)"
-            
-            let timeForSprinkle = TimePicked.date
-            let components = Calendar.current.dateComponents([.hour, .minute], from: timeForSprinkle)
-            InfoLabel.text = "Sprinkles will be sent every day at \(components.hour):\(components.minute)"
-            let notificationTime = UNCalendarNotificationTrigger(dateMatching: components, repeats: true)
-            
-        }
+    // MARK: - Private
+    private func updateUI() {
+        [DescLabel, DescButton].forEach { $0.isHidden = !Switch.isOn }
```

Comment:
So, I'm not sure you've used the `forEach` function before, but it is really useful in performing a change on a number of objects. So in this case, I made an array of `DescLabel` and `DescButton`, then I called the `forEach` function on this array.

The $0 basically is just a syntactical way of referring to the element that is being iterated over (think of forEach just like a for loop). Then I am setting the `isHidden` value for the elements to be equal to the `!Switch.isOn`, as they should be hidden when the switch is NOT on.